tests/integration/backup/self_reference/import_test.go

```diff
@@ -0,0 +1,303 @@
+// Copyright 2023 Democratized Data Foundation
+//
+// Use of this software is governed by the Business Source License
+// included in the file licenses/BSL.txt.
+//
+// As of the Change Date specified in that file, in accordance with
+// the Business Source License, use of this software will be governed
+// by the Apache License, Version 2.0, included in the file
+// licenses/APL.txt.
+
+package backup
+
+import (
+	"testing"
+
+	testUtils "github.com/sourcenetwork/defradb/tests/integration"
+)
+
+func TestBackupSelfRefImport_Simple_NoError(t *testing.T) {
+	test := testUtils.TestCase{
+		Actions: []any{
+			testUtils.BackupImport{
+				ImportContent: `{
+					"User":[
+						{
+							"_key":"bae-790e7e49-f2e3-5ad6-83d9-5dfb6d8ba81d",
+							"age":31,
+							"boss_id":"bae-e933420a-988a-56f8-8952-6c245aebd519",
+							"name":"Bob"
+						},
+						{
+							"_key":"bae-e933420a-988a-56f8-8952-6c245aebd519",
+							"age":30,
+							"name":"John"
+						}
+					]
+				}`,
+			},
+			testUtils.Request{
+				Request: `
+					query  {
+						User {
+							name
+							boss {
+								name
+							}
+						}
+					}`,
+				Results: []map[string]any{
+					{
+						"name": "Bob",
+						"boss": map[string]any{
+							"name": "John",
+						},
+					},
+					{
+						"name": "John",
+						"boss": nil,
+					},
+				},
+			},
+		},
+	}
+
+	executeTestCase(t, test)
+}
+
+// This test documents undesirable behaviour, as the document is not linked to itself.
+// https://github.com/sourcenetwork/defradb/issues/1697
+func TestBackupSelfRefImport_SelfRef_NoError(t *testing.T) {
+	test := testUtils.TestCase{
+		Actions: []any{
+			testUtils.BackupImport{
+				// Note: Whilst generating the `boss_id` field value manually would be really difficult,
+				// the document may be exported with `_key`/`_newKey` values, resulting in an export that
+				// could not be fully imported, although all the information required is there.
+				ImportContent: `{
+					"User":[
+						{
+							"_key":"bae-790e7e49-f2e3-5ad6-83d9-5dfb6d8ba81d",
```

Comment:
suggestion: Since `_newKey` will be used to check if it's a self reference, I suggest we change `_key` for `_newKey` or have both.

Reply:
> `_newKey` will be used to check if it's a self reference

It would?  I always saw it the other way round, as any exported relation id fields would contain the old key, not the new one.

Happy to have both.

- [ ] add _newKey

Reply:
The relation ID will be updated on export.

Reply:
Converted to the import-export flow suggested in another comment